test/test_expose.py

```diff
@@ -68,24 +61,21 @@ def check_namespace_properties(self, process_left, namespace_left, process_right
         port_namespace_left = process_left.spec().inputs.get_port(namespace_left)
         port_namespace_right = process_right.spec().inputs.get_port(namespace_right)
 
-        # Pop the ports in stored in the `_ports` attribute
-        port_namespace_left.__dict__.pop('_ports', None)
-        port_namespace_right.__dict__.pop('_ports', None)
+        left_dict = {k: v for k, v in port_namespace_left.__dict__.items() if k != '_ports'}
+        right_dict = {k: v for k, v in port_namespace_right.__dict__.items() if k != '_ports'}
```

Comment:
In the old implementation, the `check_namespace_properties` will change the shared state of the processes between tests, I make copies for assertion.

Reply:
Since it is test, the change is safe.